grafana/grafana
# Query History: Implement RemoteStorage methods: get all, add new

**What this PR does / why we need it**:

First implementation of the remote storage API supporting:

*  adding a new query to the history (after running it)
* getting the list of items in query history:
  * input search and data source filtering 
  * filtering by starred items cannot be tested yet as the API to star an item is not added yet (to be done in #48406)
  * limited to 100 results (pagination/load more will be implemented in #48407)
  * settings are hardcoded, to be implemented in #48408

**Which issue(s) this PR fixes**:



Fixes #47252

**Special notes for your reviewer**:

How to test it?

Enable query history:

```
[query_history]
enabled = true
```

Once query history is enabled it doesn't use local storage anymore so you start with a clean slate. Migration from local storage will be implemented in #48409.

Run a few queries and check if they are available in query history. Filtering by data source and search input should work as expected. Filtering by time, starring, commenting, deleting items, and changing settings is not supported yet.
